app/services/sso/openid_connect.rb

```diff
@@ -20,6 +22,11 @@ def authenticated?
       self.user = User.current.presence || authenticate!
     end
 
+    def logout_url
+      return Setting['login_delegation_logout_url'] if Setting['login_delegation_logout_url'].present?
```

Comment:
My reasoning to always append the `?logout` is that mod_auth_oidc requires this. By using the setting, you can accommodate a path different from `redirect_uri`. I'd argue we always want to redirect to the same URL as well to provide the same experience everywhere.

Reply:
@ekohl as mention in mod_auth_openidc(https://github.com/zmartzone/mod_auth_openidc/wiki#9-how-do-i-logout-users), the `OIDCRedirectURI` should be a protected mod_auth_openidc URL, which in our case is the /users/extlogin URL. While running the keycloak-httpd-client-install command we make the /users/extlogin a protected URL.

This `?logout` works only with the protected URL and will not work with other URLs.

Reply:
You're right that it should be a protected URL, but `redirect_uri` is configurable. See https://github.com/theforeman/foreman/pull/7245#discussion_r358207123 as well.

Reply:
@ekohl the reason I don't want to append `?logout` to the `Setting['login_delegation_logout_url']` is because if I do so, it will log me out of mod_auth_openidc also, this will inturn result in mod_auth_openidc going to Keycloak and logging me out from there too. Which we don't want as @ares mentioned. 

That is, we want to logout from Foreman and not SSO, but if I append `?logout=` to the `Setting['login_delegation_logout_url']`  it will log me out from the SSO also.

My comment https://github.com/theforeman/foreman/pull/7245#issuecomment-566650797 explains the currect functionality. Let me know if you find it correct.